pkg/registry/generic/rest/proxy.go

```diff
@@ -130,33 +137,35 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 	if !httpstream.IsUpgradeRequest(req) {
 		return false
 	}
-
-	backendConn, err := proxy.DialURL(h.Location, h.Transport)
-	if err != nil {
+	if err := h.handleUpgrade(w, req); err != nil {
```

Comment:
This change moves the hijack and close of the request connection before we get a chance to respond to errors dialing the backend. That means we would no longer send API error responses in error cases.